### ISSUE TITLE
Which is the loss function used for multi-class classification?

### ISSUE THREAD

**Issue by AnabelGRios** [dmlc/xgboost#2024]
Hello!

I'm trying to know which is the loss function used for multi-class classification. I had though it might be the same as in GBM (for K classes):
![codecogseqn 1](https://cloud.githubusercontent.com/assets/8157671/22786524/2a1d6da4-eed9-11e6-9a85-0359d042019c.png)
where y_k=1 if x's label is k and 0 in any other case, and p_k(x) is the softmax function. However, I have made the first and second order gradient using this loss function and the hessian doesn't match the one defined in the code [here](https://github.com/dmlc/xgboost/blob/master/src/objective/multiclass_obj.cc) by a constant 2.

Could you please tell me which is the loss function used?

Thank you!

**Comment by marugari** [comment]
See https://github.com/dmlc/xgboost/issues/638

I'm also interested in this.

**Comment by tqchen** [comment]
https://github.com/dmlc/xgboost/issues/1825
